### ISSUE TITLE
Link creation order not stored for file

### ISSUE THREAD

**Issue by MarDiehl** [HDFGroup/hdf5#5183]
**Bug**
The information whether a file was created with `h5p_crt_order_indexed` and `h5p_crt_order_tracked` is not (or only partially) stored. This means that `H5Pget_link_creation_order` does not report those values to be set if a file is re-opened. For groups, the information is stored. Note that HDFView still sorts the data in the file/base_group if it was created with order_tracked/order_indexed but in agreement with the Fortran library reports that order was not tracked.

**Expected behavior**
I would expect the same behavior as for groups: If a file was created with with `h5p_crt_order_indexed` and `h5p_crt_order_tracked`, this information should be stored and `H5Pget_link_creation_order` reports that the flags are set when opening the file again later.

**Platform (please complete the following information)**
 - HDF5 version: 1.14.5
 - System: Arch linux (current)
 - Compiler: GCC 14.2.1.
 - Build: Official package (mix of autotool/CMake)
 - Any configure options you specified: see https://archlinux.org/packages/extra/x86_64/hdf5-openmpi
 - MPI library and version (parallel HDF5): openMPI

**Additional context**
Initially, I thought that this was an HDFView issue and opened https://github.com/HDFGroup/hdfview/issues/347

***Code to Reproduce***

[sort_file_f90.txt](https://github.com/user-attachments/files/18165231/sort_file_f90.txt): run to create 2 files
[print_file_py.txt](https://github.com/user-attachments/files/18165234/print_file_py.txt): print order in Python
[get_status_file_f90.txt](https://github.com/user-attachments/files/18165233/get_status_file_f90.txt): get status in Fortran

[sort_group_f90.txt](https://github.com/user-attachments/files/18165232/sort_group_f90.txt): run to create 2 groups
[print_group_py.txt](https://github.com/user-attachments/files/18165236/print_group_py.txt): print order in Python
[get_status_group_f90.txt](https://github.com/user-attachments/files/18165235/get_status_group_f90.txt): get status in Fortran

**Comment by fortnern** [comment]
Does this problem occur if you only use one language? If so would you be able to send a reproducer in a single language (ideally C)?

**Comment by MarDiehl** [comment]
yes, it occurs for Fortran only. Writing and reading to file is just split over two programs. Unfortunately, my C skills are not that good, but I can merge the code into one reproducer if that helps.

**Comment by MarDiehl** [comment]
Here's a small reproducer. I compile with
```fortran
gfortran write_file_and_check.f90 -lhdf5_fortran -I/usr/include
```

[write_file_and_check_f90.txt](https://github.com/user-attachments/files/18170217/write_file_and_check_f90.txt)

**Comment by brtnfld** [comment]
Thank you for the report and reproducer; they were very helpful. I've confirmed this is a bug in the C library, but it was not because the information was not stored in the file. If you would like a workaround, you can instead do:

```
 CALL H5Fopen_f(filename, H5F_ACC_RDWR_F, file_id, error)
 CALL H5Gopen_f(file_id, ".", group_id, error)
 CALL H5Gget_create_plist_f(group_id, plist_id, error)
 CALL H5Pget_link_creation_order_f(plist_id, corder,error)
```

**Comment by MarDiehl** [comment]
> Thank you for the report and reproducer; they were very helpful. I've confirmed this is a bug in the C library ..

Thank you. For picking this up and HDF5 in general!
